packages/perseus/src/components/highlighting/ui/highlight-tooltip.jsx

```diff
@@ -30,6 +30,15 @@ type HighlightTooltipProps = {|
     onMouseEnter?: () => mixed,
     onMouseLeave?: () => mixed,
 
+    // This is only to be used by tests and exists because
+    // tests don't seem to play nicely with building a highlight
+    // range in `_getFocusRect`
```

Comment:
I'm pretty averse to extra props just for testing.  `document.createRange()` looks to be supported by JSDom. 🤔  I'd be curious to see what's not working or breaking.

Reply:
If it's DOM measurement APIs like `getBoundingClientRect`, you might look at `config/test/attach-jsdom-window-shims.js` and maybe we can shim in any missing APIs enough to get this working with the production code unchanged.